src/cipher.rs

```diff
@@ -509,102 +511,296 @@ impl rustls_root_cert_store {
     }
 }
 
+/// A builder for a `rustls_allow_any_authenticated_client_verifier`. This builder object can be
+/// used to configure certificate revocation lists, and then turned into a
+/// `rustls_allow_any_authenticated_client_verifier` once ready.
+pub struct rustls_allow_any_authenticated_client_builder {
+    _private: [u8; 0],
+}
+
+impl CastPtr for rustls_allow_any_authenticated_client_builder {
+    // XXX: contained value is consumed even on error, so this can contain None. but the caller
+    // still needs to free it
+    type RustType = Option<AllowAnyAuthenticatedClient>;
+}
+
+impl BoxCastPtr for rustls_allow_any_authenticated_client_builder {}
+
+impl rustls_allow_any_authenticated_client_builder {
+    /// Create a new allow any authenticated client certificate verifier builder using the root store.
+    ///
+    /// This copies the contents of the rustls_root_cert_store. It does not take
+    /// ownership of the pointed-to memory.
+    ///
+    /// This object can then be used to load any CRLs.
+    ///
+    /// Once that is complete, convert it into a real `rustls_allow_any_authenticated_client_verifier`
+    /// by calling `rustls_allow_any_authenticated_client_verifier_new()`.
+    #[no_mangle]
+    pub extern "C" fn rustls_allow_any_authenticated_client_builder_new(
+        store: *const rustls_root_cert_store,
+    ) -> *mut rustls_allow_any_authenticated_client_builder {
+        ffi_panic_boundary! {
+            let store: &RootCertStore = try_ref_from_ptr!(store);
+            let client_cert_verifier = Some(AllowAnyAuthenticatedClient::new(store.clone()));
+            BoxCastPtr::to_mut_ptr(client_cert_verifier)
+        }
+    }
+
+    /// Add one or more certificate revocation lists (CRLs) to the client certificate verifier by
+    /// reading the CRL content from the provided buffer of PEM encoded content.
+    ///
+    /// This function may return an error if the provided buffer is not valid PEM encoded content,
+    /// or if the CRL content is invalid or unsupported.
+    #[no_mangle]
+    pub extern "C" fn rustls_allow_any_authenticated_client_builder_add_crl(
+        builder: *mut rustls_allow_any_authenticated_client_builder,
+        crl_pem: *const u8,
+        crl_pem_len: size_t,
+    ) -> rustls_result {
+        ffi_panic_boundary! {
+            let client_cert_verifier_builder: &mut Option<AllowAnyAuthenticatedClient> = try_mut_from_ptr!(builder);
+
+            let crl_pem: &[u8] = try_slice!(crl_pem, crl_pem_len);
+            let crls_der = match crls(&mut Cursor::new(crl_pem)) {
+                Ok(vv) => vv.into_iter().map(UnparsedCertRevocationList).collect::<Vec<UnparsedCertRevocationList>>(),
```

Comment:
```suggestion
            let crls_der: Vec<UnparsedCertRevocationList> = match crls(&mut Cursor::new(crl_pem)) {
                Ok(vv) => vv.into_iter().map(UnparsedCertRevocationList).collect(),
```

This is a stylistic thing, but I prefer type annotations on the variable declaration over turbofish. Some reasons:

 - Turbofish involves a lot of punctuation, including extra nesting of `<>`, which is hard to read.
 - Turbofish usually winds up in a nested expression, and so contributes to rightward pressure and line wrapping.
  - As long as we're explicitly writing out the type rather than allowing it to be inferred, it's better to write it in one consistent place up front. Otherwise, the reader has to look through potentially many nested branches to find the turbofish operator.

This is slightly aside from the question of how frequently to use type annotations on variable declarations _in general_, a matter on which I know opinions differ. :D

There's one other case of this in `rustls_client_cert_verifier_optional_builder_add_crl` on line 712.

Reply:
>This is a stylistic thing, but I prefer type annotations on the variable declaration over turbofish.

I switched this over, but also want to mention that this preference is opposite the feedback I've been getting on my rustls + webpki PRs. As you note I think it often comes down to a stylistic choice but the arguments you make in favour of explicit annotations seem sensible to me. 

I don't have a strong preference one way or the other. It might be something we should try to hash out in a `CONTRIBUTING.md` update over in the main Rustls repo to try and get some consistency and to avoid new contributors having to flip/flop based on who reviews their PR first :-) I've been collecting up some other notes I'd like to turn into a PR at some point. I can tag you when that happens if you'd like.